mesonbuild/backend/ninjabackend.py

```diff
@@ -3611,6 +3615,12 @@ def generate_ending(self):
         elem.add_item('pool', 'console')
         self.add_build(elem)
 
+        # If these files used to be explicitly created, they need to appear on the build graph somehow,
+        # otherwise cleandead deletes them. See https://github.com/ninja-build/ninja/issues/2299
+        if self.implicit_meson_outs:
+            elem = NinjaBuildElement(self.all_outputs, 'meson-internal__stamp', 'phony', self.implicit_meson_outs)
```

Comment:
`meson-internal__stamp` name does not mean much, maybe call it `meson-internal-implicit-outs` to  have  name that match what we used in the meson code?

Reply:
Other than this naming nitpicking, LGTM.

Reply:
Fair enough.. actually we should also remove the internal__ since that is used for namespacing alias_target.